### ISSUE TITLE
Upgrade to neorg v8

### ISSUE THREAD

**Issue by Toraja** [Toraja/toybox#45]
`neorg` v8 requires `luarocks.nvim` but it cannot be installed as-is if you install lua via `asdf`.   

Required change:
```diff
diff --git a/lua/luarocks-nvim/build.lua b/lua/luarocks-nvim/build.lua
index 71c7d8d..5dd2ec4 100644
--- a/lua/luarocks-nvim/build.lua
+++ b/lua/luarocks-nvim/build.lua
@@ -118,8 +118,9 @@ local steps = {
                                "sh",
                                "configure",
                                "--prefix=" .. paths.rocks,
-                               "--lua-version=5.1",
+                               "--lua-version=5.4",
                                "--force-config",
+                               "--with-lua-include=" .. vim.loop.os_homedir() .. "/.asdf/installs/lua/5.4.6/include",
                        }
                        vim.list_extend(cmd, luarocks_args or {})
                        local error_code, stdout, stderr = run_job(cmd)
```

With the above change, `Lazy build luarocks.nvim` succeeds, but when reopening nvim, an error occurs saying: 
```
Unable to load the luarocks package loader. Try re-running the build stage (`:Lazy build luarocks.nvim`).
```

Rebuilding did not help.  

I guess I should install lua using `apt` or change the lua version to `5.1` as it seems that `luarock.nvim` only supports lua 5.1.

**Comment by Toraja** [comment]
With lua `5.1.5`, this works and upgrading neorg to v8 succeeded
```diff
diff --git a/lua/luarocks-nvim/build.lua b/lua/luarocks-nvim/build.lua
index 71c7d8d..eacf60a 100644
--- a/lua/luarocks-nvim/build.lua
+++ b/lua/luarocks-nvim/build.lua
@@ -125,6 +125,7 @@ local steps = {
                        local error_code, stdout, stderr = run_job(cmd)
                        -- some known workarounds for better user experience
                        local next_try_args = {
+                               { "--with-lua-include=" .. vim.loop.os_homedir() .. "/.asdf/installs/lua/5.1.5/include" },
                                { "--with-lua-include=/usr/include" }, -- arch linux
                        }
                        -- while error_code ~= 0 try the workarounds
```

**Comment by Toraja** [comment]
neorg (https://github.com/nvim-neorg/neorg/commit/a8b7ad08f6e8cd36c83e371ab9a74cc1e5252a0b) uses nvim feature that is not available as of v0.9.5 (currently latest).  
Uncomment the below.
```diff
diff --git a/lua/neorg/modules/core/todo-introspector/module.lua b/lua/neorg/modules/core/todo-introspector/module.lua
index 52c24fee..f0dc8344 100644
--- a/lua/neorg/modules/core/todo-introspector/module.lua
+++ b/lua/neorg/modules/core/todo-introspector/module.lua
@@ -177,7 +177,7 @@ function module.public.perform_introspection(buffer, node)
                 "Normal",
             },
         },
-        invalidate = true,
+        -- invalidate = true,
     })
 end
```

**Comment by Toraja** [comment]
Closing as it is now working properly
